web/src/engine/websites/templates/FlatManga.ts

```diff
@@ -0,0 +1,110 @@
+﻿import { AddAntiScrapingDetection, FetchRedirection } from "../../platform/AntiScrapingDetection";
+import { FetchCSS } from "../../platform/FetchProvider";
+import { DecoratableMangaScraper, type Chapter, type MangaScraper } from "../../providers/MangaPlugin";
+import { Page } from "../../providers/MangaPlugin";
+import * as Common from '../decorators/Common';
+
+AddAntiScrapingDetection(async (invoke) => {
+    const result = await invoke<boolean>(`document.documentElement.innerHTML.includes('ct_anti_ddos_key')`);// Sample => Mangagun, NicoManga, Rawinu, Weloma, WeloveManga
+    if (result) {
+        await new Promise(resolve => setTimeout(resolve, 3000));
+        return FetchRedirection.Automatic;
+    } else return undefined;
+
+});
+
+export function MangaLabelExtractor(element: HTMLElement) {
+    return CleanTitle(element.getAttribute('text') ? element.getAttribute('text') : element.textContent);
+}
+
+export function MangaExtractor(anchor: HTMLAnchorElement) {
+    return {
+        id: anchor.pathname,
+        title: CleanTitle(anchor.getAttribute('text') ? anchor.text : anchor.textContent)
+    };
+}
+
+export function CleanTitle(title: string): string {
+    title = title.replace(/\s*[~\-―〜]\s*RAW\s*\(MANGA\)\s*$/i, '').trim();
+    title = title.replace(/\s*[~\-―〜]\s*(MANGA)?\s*RAW\s*$/i, '').trim();
+    title = title.replace(/\(raw\)/i, '').trim();
+    return title.replace(/\(manga\)/i, '').trim();
+}
+
+export function PageLinkExtractor<E extends HTMLImageElement>(this: MangaScraper, element: E): string {
+    let page = element.dataset.aload || element.dataset.src || element.dataset.srcset || element.dataset.original || element.dataset.pagespeedLazySrc || element.src;
```

Comment:
The `dataset.aload` and `dataset.pagespeedLazySrc` seems to be very website specific, should these really be part of the template?